src/lib/server/index.ts

```diff
@@ -12,12 +12,145 @@ let db: DB | null = null;
+                .values(v.splice(0, Math.min(100, v.length)))
+                .returning()
+        );
+    }
+
+    const channels = await db
+        .insert(schema.channelTable)
+        .values(
+            users.slice(0, 20).map((u) => ({
+                name: rand_username(),
+                createdBy: u.id,
+            }))
+        )
+        .returning();
+
+    await db.insert(schema.publicChannelTable).values(
+        channels.map((c) => ({
+            name: c.name,
+            channelId: c.id,
+        }))
+    );
+
+    const tags: { [id: string]: schema.ChannelTag[] } = {};
+    for (const channel of channels) {
+        const t = pick_n(words, 5);
+        tags[channel.id] = await db
+            .insert(schema.channelTagsTable)
+            .values(
+                Array(5)
+                    .fill(0)
+                    .map((_, i) => ({
+                        channelId: channel.id,
+                        name: t[i],
+                        color: `#${rand(255).toString(16)}${rand(255).toString(16)}${rand(255).toString(16)}`,
+                    }))
+            )
+            .returning();
+    }
+
+    const video_ids = [
+        'e0245338-7c04-4a6c-b44f-0e279a849cf5',
+        'ee59e892-7838-46d1-876d-49efb4feb7ba',
+        '6df184ad-6175-4dbb-88b0-7bb83c88e74c',
+        '9dae1ca5-48e8-4fed-81e7-90dff6ca6d51',
+        'ba0aa315-365c-4f90-b72a-3ddacee81381',
+        '0164c7a6-2b48-4c96-8e02-34907666ec77',
+    ];
+    const posts = await db
+        .insert(schema.postTable)
+        .values(
+            Array(100)
+                .fill(0)
+                .map(() => ({
+                    title: lip(rand(10, 2)),
+                    channelId: channels[rand(channels.length)].id,
+                    createdBy: users[rand(users.length)].id,
+                    videoId: video_ids[rand(video_ids.length)],
+                }))
+        )
+        .returning();
+
+    for (const post of posts) {
+        const t = [...tags[post.channelId]];
+        const fill = Array(rand(5, 1))
+            .fill(0)
+            .map(() => {
+                const [tag] = t.splice(rand(t.length), 1);
+                return {
+                    postId: post.id,
+                    tagId: tag.id,
+                };
+            });
+        if (fill.length) {
+            await db.insert(schema.postTagTable).values(fill).returning();
+        }
+    }
+
+    console.log('user count:', users.length);
+    console.log('channel count:', channels.length);
+    console.log('posts count:', posts.length);
+
+    for (const user of users) {
+        const vote_v = [];
+        for (const post of posts) {
+            vote_v.push({
+                postId: post.id,
+                userId: user.id,
+                vote: Math.random() > 0.25 ? ('UP' as const) : ('DOWN' as const),
+            });
+        }
+        await db.insert(schema.postVoteTable).values(vote_v);
+    }
+    console.log('generated data');
+};
+
```

Comment:
nit: I love this. I'd prefer if it were in some sort of utilities file though. It doesn't really belong in index imo.